nench.sh

```diff
@@ -185,12 +186,15 @@ uname -s -r -m
 printf '\n'
 
 printf 'Disks:\n'
-if command_exists lsblk
+if command_exists lsblk && [ -n "$(lsblk)" ]
 then
     lsblk --nodeps --noheadings --output NAME,SIZE,ROTA --exclude 1,2,11 | sort | awk '{if ($3 == 0) {$3="SSD"} else {$3="HDD"}; printf("%-3s%8s%5s\n", $1, $2, $3)}'
 elif [ -r "/var/run/dmesg.boot" ]
 then
     awk '/(ad|ada|da|vtblk)[0-9]+: [0-9]+.B/ { print $1, $2/1024, "GiB" }' /var/run/dmesg.boot
+elif command_exists df
+then
+    df -h --output=source,fstype,size,itotal | awk 'NR == 1 || /^\/dev/'
```

Comment:
This will include non-physical devices (`tmpfs`, NFS, Overlays, …).
I'd prefer to have a way to exclude those before merging this change.

Do you have a concrete case where the other two commands failed and the `df` solution was required?

Reply:
ovz

Reply:
Ah, I missed your `awk` filter. Should be fine then.